CMakeLists.txt

```diff
@@ -353,13 +390,28 @@ protobuf_generate_cpp(PROTO_SRCS PROTO_HDRS ${PROTO_FILES})
 message("pika PROTO_SRCS = ${PROTO_SRCS}")
 message("pika PROTO_HDRS = ${PROTO_HDRS}")
 
-add_executable(${PROJECT_NAME}
+add_library(${PROJECT_NAME}_static
```

Comment:
our purpose is to have a pika executable binary which not depend on system package, so  a pika_static is really necessary？

Reply:
Now all the libraries are linked as static, the pika executable binary don't need to depend on other package (except libprotobuf which is linked as shared)